Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -1403,6 +1403,21 @@ - (NSDictionary *) parseFieldDefinitionStringParts:(NSArray *)definitionParts
 			}
 			definitionPartsIndex++;
 
+        // Whether field is generated always as xxxx (VIRTUAL or STORED)
+        } else if ([detailString isEqualToString:@"GENERATED"] && (definitionPartsIndex + 4 < partsArrayLength)
+                    && [[[definitionParts safeObjectAtIndex:definitionPartsIndex+1] uppercaseString] isEqualToString:@"ALWAYS"]
+                    && [[[definitionParts safeObjectAtIndex:definitionPartsIndex+2] uppercaseString] isEqualToString:@"AS"]) {
+            if ([[aValue = [definitionParts safeObjectAtIndex:definitionPartsIndex+4] uppercaseString] isEqualToString:@"VIRTUAL"]) {
+                [fieldDetails setValue:aValue forKey:@"generatedalways"];
```

Comment:
Why not make the key something like `generatedType` and storing either `VIRTUAL` or `STORED` depending on which type it is? I'm thinking that knowing whether it's virtual or stored could be useful info in the future (there could of course be some other special thing in the future) and should be just as useable to do all the checks

Reply:
It's already the case. The string `VIRTUAL` or `STORED` (var `aValue`) is storing with the key `generatedType`.

Reply:
Ahhh I see